File: core/src/main/java/by/mrrockka/domain/MoneyTransfer.java
package by.mrrockka.domain;

import by.mrrockka.domain.payout.TransferType;
import lombok.NonNull;

import java.math.BigDecimal;
import java.util.UUID;

public record MoneyTransfer(
  @NonNull
  UUID personId,
  @NonNull
  UUID gameId,
  @NonNull
  BigDecimal amount,
  @NonNull
  TransferType type
) {
}

File: core/src/main/java/by/mrrockka/domain/MoneyTransfer.kt
package by.mrrockka.domain

import java.math.BigDecimal
import java.util.*

data class MoneyTransfer(
        val personId: UUID,
        val gameId: UUID,
        val amount: BigDecimal,
        val type: TransferType,
)

enum class TransferType {
    DEBIT, CREDIT, EQUAL
}
